### ISSUE TITLE
Add Unit Tests for Model Components to Ensure Correctness

### ISSUE THREAD

**Issue by porkfriday** [kwuking/TimeMixer#140]
**Description:**
The repository currently lacks automated tests, which makes it difficult to verify the correctness of model components and ensure that changes don't break existing functionality. Adding unit tests would improve code reliability and make the codebase more maintainable.

**Current State:**
- No automated tests for model components
- Changes to core model architecture could introduce subtle bugs
- Difficult to verify correctness of individual components

**Proposed Solution:**
Implement a comprehensive test suite using PyTest that includes:

1. Unit tests for key model components:
   - `DFT_series_decomp`
   - `MultiScaleSeasonMixing`
   - `MultiScaleTrendMixing`
   - `PastDecomposableMixing`
   - `Model` forward passes

2. Test cases should cover:
   - Input/output shapes
   - Edge cases (e.g., single sample, minimum sequence length)
   - Expected behavior with different configurations
   - Gradient flow during backpropagation

**Example Test:**
```python
import pytest
import torch
from models.TimeMixer import DFT_series_decomp

def test_dft_decomposition():
    # Setup
    batch_size, seq_len, dim = 2, 96, 7
    x = torch.randn(batch_size, seq_len, dim)
    decomp = DFT_series_decomp(top_k=5)
    
    # Execute
    season, trend = decomp(x)
    
    # Verify
    assert season.shape == x.shape
    assert trend.shape == x.shape
    # Sum of seasonal and trend should approximately equal the original
    assert torch.allclose(season + trend, x, rtol=1e-5, atol=1e-5)
    
    # Verify reconstruction property
    recon = season + trend
    assert torch.allclose(recon, x, rtol=1e-5, atol=1e-5)
```

**Benefits:**
- Early detection of bugs during development
- Improved confidence when making changes to the codebase
- Documentation of expected component behavior
- Facilitates contribution from other developers
- Makes the codebase more reliable and maintainable

**Additional Suggestion:**
Consider integrating the tests with CI/CD to automatically run tests on pull requests.
